src/main/java/com/lambdaschool/oktafoundation/services/PartnerApplicationServiceImpl.java

```diff
@@ -0,0 +1,168 @@
+
+
+
+
+        return partnerapprepos.save(newPartApp);
+    }
+
+    @Transactional
+    @Override
+    public PartnerApplication update(User user, PartnerApplication partnerApplication, long id)
+    {
+        User currentUser = user;
+
+        PartnerApplication currentPartApp = partnerapprepos.findById(id)
+            .orElseThrow(() -> new ResourceNotFoundException("Application id " + id + " not found!"));
+
+        if (partnerApplication.getName() != null)
+        {
+            currentPartApp.setName(partnerApplication.getName());
+        }
+        if (partnerApplication.getAddress() != null)
+        {
+            currentPartApp.setAddress(partnerApplication.getAddress());
+        }
+
+        if (partnerApplication.getPhone() != null)
+        {
+            currentPartApp.setPhone(partnerApplication.getPhone());
+        }
+
+        if (partnerApplication.getType() != null)
+        {
+            currentPartApp.setType(partnerApplication.getType());
+        }
+
+        if (partnerApplication.getDescription() != null)
+        {
+            currentPartApp.setDescription(partnerApplication.getDescription());
+        }
+        //since users won't update this then we can either leave out or reset it here
+        if (partnerApplication.getStatus() != null)
+        {
+            currentPartApp.setPhone(partnerApplication.getPhone());
+        }
+
+        return partnerapprepos.save(currentPartApp);
+    }
+
+    @Transactional
+    @Override
+    public PartnerApplication updatePartAppStatus(
+        PartnerApplication partnerApplication,
+        long id)
+    {
+        PartnerApplication currentApp = partnerapprepos.findById(id)
+            .orElseThrow(() -> new ResourceNotFoundException("Application id " + id + " not found!"));
+
+        // user can check status
+        if (partnerApplication.getStatus() != null)
+        {
+            currentApp.setStatus(partnerApplication.getStatus());
+        }
+
+        // many to one user
+        currentApp.setUser(partnerApplication.getUser());
+
+        return partnerapprepos.save(currentApp);
+    }
+
+    @Override
+    public List<PartnerApplication> findAll()
+    {
+        List<PartnerApplication> list = new ArrayList<>();
+
+        partnerapprepos.findAll().iterator().forEachRemaining(list::add);
+        return list;
+    }
+
+    @Override
+    public PartnerApplication findPartAppById(long id)
+    {
+        return partnerapprepos.findById(id)
+            .orElseThrow(() -> new ResourceNotFoundException("Application id " + id + " not found!"));
+    }
+
+    @Transactional
+    @Override
+    public PartnerApplication saveByAuth(User user, PartnerApplication partnerApplication)
+    {
+        PartnerApplication newPartApp = new PartnerApplication();
+//        User u = userrepos.findById(user.getUserid())
+//            .orElseThrow(() -> new ResourceNotFoundException("User id " + user.getUserid() + " not found!"));
+//        newPartApp.setUser(u);
```

Comment:
This is pretty much the same as my other comment. We should be able to just let this code execute by un-commenting it.